internal/deployment/deployment.go

```diff
@@ -104,9 +104,15 @@ func FromFile(path util.AbsolutePath) (*Deployment, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Migration
+	if d.DashboardURL == "" && d.ID != "" {
+		d.DashboardURL = util.GetDashboardURL(d.ServerURL, d.ID)
+	}
+	if d.DirectURL == "" && d.ID != "" {
+		d.DirectURL = util.GetDirectURL(d.ServerURL, d.ID)
+	}
 	if d.LogsURL == "" && d.DashboardURL != "" {
-		// Migration
-		d.LogsURL = util.URLJoin(d.DashboardURL, "logs")
+		d.LogsURL = util.GetLogsURL(d.ServerURL, d.ID)
```

Comment:
Is the migration necessary here? If I'm understanding correctly pre-deployments didn't have `id` and full deployments did. In the full deployment case the URLs were always provided right?

It looks like the update to `patch_deployment` also writes the URLs.

Reply:
No, you're right. I do not think the migration is needed at this time. Mike placed the original case in there because we didn't always have the logs URL present in deployment files, but my changes are going in as one merge, so we won't have the issue of having predeployment files with the ID but without the URLs. (I had them within my development environment because of the incremental development process.)

Thanks for pointing this out. I'll remove my migration and make a minor change to what Mike had there before.